spec/bin/query-cloudwatch_spec.rb

```diff
@@ -292,6 +325,237 @@ def build_stdin_with_query(query)
       end
     end
 
+    context 'with sqlite format' do
+      let(:format) { :sqlite }
+
+      let(:db) do
+        SQLite3::Database.new(':memory:')
+      end
+
+      before do
+        allow_any_instance_of(QueryCloudwatch::SqliteOutput).to receive(:db).
+          and_return(db)
+        allow_any_instance_of(QueryCloudwatch::SqliteOutput).to receive(:close_database)
```

Comment:
It gets kind of tricky because `stdout` and `stderr` are passed into `run` and you're creating a new output instance for each run invocation bound to those values.